lib/carrierwave/mongoid.rb

```diff
@@ -255,4 +342,24 @@ class CarrierWave::Uploader::Base
   end
 end
 
+if Gem::Version.new("3.0.beta") <= Gem::Version.new(CarrierWave::VERSION) &&
+  Gem::Version.new(CarrierWave::VERSION) < Gem::Version.new("3.1.1")
+  # Monkey patch to work around https://github.com/carrierwaveuploader/carrierwave/issues/2770
+  CarrierWave::SanitizedFile.prepend Module.new {
+    def read(*args)
+      if args.empty?
+        super
+      else
+        if is_path?
+          @file = File.open(path, "rb")
+        elsif @file.is_a?(CarrierWave::Uploader::Base)
+          @file = StringIO.new(@file.read)
+        end
+
+        @file.read(*args)
+      end
+    end
+  }
+end
```

Comment:
carrierwave-mongoid was heavily affected by the issue in carrierwave: https://github.com/carrierwaveuploader/carrierwave/issues/2770. The issue is hard to investigate due to its nature, and affected many carrierwave versions (from 3.0.0 to 3.1.0). Hence I've added a monkey patch here, to ensure carrierwave-mongoid works with flawed carrierwave versions.